### PR TITLE
Closes #876: Enable testing of related az-quickstart-dev branches in main repo CI builds.

### LINKED ISSUE #876: Enable testing of related az-quickstart-dev (require-dev metapackage) branches in main repo CI builds
## Motivation
We currently have provisions in our ProboCI, lando, ddev and GH actions configs to allow changes in both our main profile repo and our scaffolding repo to be tested simultaneously if the same branch name exists in both repos.

It would be useful to have the same kind of provisions enabled for our `az-quickstart-dev` (require-dev composer metapacakge) as well.

## Proposed Resolution
Add steps to CI configs for checking to use matching branch name if one exists in `az-quickstart-dev`.

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Enable CI builds to use matching branch name in az-quickstart-dev repo if it exists.

## Related Issue
#876 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — COMMENTED** [review]
are you sure this is just a draft? looks good to me.

**Comment by joeparsons** [comment]
> are you sure this is just a draft? looks good to me.

Still need to add lando and ddev support.

**Comment by joeparsons** [comment]
Tasks remaining:
- [ ] Update lando config
- [ ] Update ddev config

**Review by kevcooper — APPROVED** [review]
Checked with lando:

1. Cloned this repo
2. Checked out `issue/876`
3. Created a new branch named `issue/1180` (This branch exists in az-quickstart-dev)
4. Ran `lando start`
5. Observed that proper version of `mglaman/phpstan-drupal` is being installed
  - `Installing mglaman/phpstan-drupal (1.1.4): Extracting archive`

**Review by trackleft — APPROVED** [review]

**Comment by trackleft** [comment]
This worked with this PR https://github.com/az-digital/az-quickstart-dev/pulls

**Comment by joeparsons** [comment]
I forgot we hadn't merged this yet.  We should do so, IMHO.

**Comment by trackleft** [comment]
I think this should be fine to merge.

File: .lando.yml [hunk 1]
```diff
@@ -23,6 +23,8 @@ services:
       # Use the matching scaffolding branch if available.
       - BRANCH_NAME=$(git -C /usr/local/quickstart-install-profile rev-parse --abbrev-ref HEAD); if [ $(git ls-remote --heads https://github.com/az-digital/az-quickstart-scaffolding.git $BRANCH_NAME | wc -l) = 1 ]; then SCAFFOLD_BRANCH="$BRANCH_NAME"; else SCAFFOLD_BRANCH="main"; fi; git clone --branch $SCAFFOLD_BRANCH https://github.com/az-digital/az-quickstart-scaffolding.git /app
       - composer config repositories.localdev path /usr/local/quickstart-install-profile && composer require --no-update drupal/core-recommended:* zaporylie/composer-drupal-optimizations:* az-digital/az_quickstart:\*@dev
+      # Use the matching require-dev metapackage branch if available.
+      - BRANCH_NAME=$(git -C /usr/local/quickstart-install-profile rev-parse --abbrev-ref HEAD); if [ $(git ls-remote --heads https://github.com/az-digital/az-quickstart-dev.git $BRANCH_NAME | wc -l) = 1 ]; then DEV_PACKAGE_BRANCH="$BRANCH_NAME"; else DEV_PACKAGE_BRANCH=main; fi; composer require --no-update --dev az-digital/az-quickstart-dev:dev-${DEV_PACKAGE_BRANCH}
```
[the remark this answers is not in the document]
Comment (trackleft) [reply .lando.yml:27]: Probably not ideal to reuse the same env variable.

**Pull request merged**